connectors/sources/google_drive.py

```diff
@@ -1092,14 +1088,12 @@ async def prepare_file(self, client, file, paths, is_shared_drive_file):
                         self._logger.error(exception_log_msg)
 
             file_document[ACCESS_CONTROL] = self._process_permissions(permissions)
-        if is_shared_drive_file:
+        if file_document.get("shared_drive"):
             return file_document, file.get("trashedTime", "")
         else:
-            return file_document
+            return file_document, ""
```

Comment:
Also perhaps worth leaving a comment what's going on in here :) 
- only shared drives can populate `trashedTime`
- personal drives never populate trashedTime

Reply:
> Why empty string and not None to indicate non existent trashed time? I think using None would be a better practice here

because if we change from "" -> None
it will cause problems in the below elif condition and raise `TypeError: '>' not supported between instances of 'NoneType' and 'str'`
https://github.com/elastic/connectors/blob/ef64cafc431ad7d9208622f9c185d493d8e3a0f6/connectors/sources/google_drive.py#L1276-L1278
If we keep it as None then we need to check if the `trashedTime is not None` and so did this.

Reply:
@moxarth-elastic I believe we can simply change the order in the `or` statement:

```python
if (trashedTime is None or trashedTime > self.last_sync_time()) and file.get("trashed") is True:
```

If `trashedTime is None` evaluates to `True,` the inequality won’t be evaluated because the or operator in Python short-circuits. It evaluates the first expression and, if it’s true, does not evaluate the second.

There’s nothing wrong with your logic. I just don’t like using “magic” empty strings for handling timestamps and inequalities. Using None is more straightforward and clear.

Reply:
Yes, I've updated the approach as per your suggestion 👍